### PR TITLE
build: enable type check on build

### PR DESCRIPTION
**Pull request by arun3528**
# COMPLETES #< INSERT LINK TO ISSUE >

## This pull request addresses

Adds type check during the build process , also runs some basic checks on js files as well 

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [x] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by maxinteger — APPROVED** [review]
Looks good

File: config/tsconfig.typecheck.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "extends": "../tsconfig.json",
+    "include": [
+        "../packages/**/*.*"
+    ],
+    "compilerOptions": {
+        "emitDeclarationOnly": false,
+    },
+    "exclude": [
+        "../packages/**/dist/**/*.*",
+        // some packages have internal reference to file so cannot exclude them indivdually
```
Comment (marcin-bazyl) [inline config/tsconfig.typecheck.json:11]: as far as I know comments like that are not allowed in json files

**Review by Neeraj-swarnkar — COMMENTED** [review]
Looks Good

File: config/tsconfig.typecheck.json (hunk 1 above)
Comment (arun3528) [reply config/tsconfig.typecheck.json:11]: i think due to the lack of the eslint rule it did not show up , will make sure next time

**Pull request merged**